src/TestSuite/IntegrationTestCase.php

```diff
@@ -424,8 +424,7 @@ protected function _buildRequest($url, $method, $data)
     protected function _addTokens($url, $data)
     {
         if ($this->_securityToken === true) {
-            $keys = Hash::flatten($data);
-            $tokenData = $this->_buildFieldToken($url, array_keys($keys));
+            $tokenData = $this->_buildFieldToken($url, array_keys($data));
```

Comment:
Wouldn't your changes would make it so only the first level of array keys are included in the token? I think that will cause data with nested fields to fail.